sql/scripts/SchemaSpy.sh

```diff
@@ -12,4 +13,41 @@ if [ ! -e "postgresql-9.4-1202.jdbc41.jar" ]; then
   wget https://jdbc.postgresql.org/download/postgresql-9.4-1202.jdbc41.jar;
 fi
 
-java -jar schemaSpy.jar -t pgsql -host $4 -s bookbrainz -db $3 -u $1 -p $2 -o _build -dp postgresql-9.4-1202.jdbc41.jar;
+while getopts "d:h:p:u:" opt; do
+  case $opt in
+    d)  DATABASE=$OPTARG;;
+    h)  HOST=$OPTARG;;
+    p)  PASSWORD=$OPTARG;;
+    u)  _USERNAME=$OPTARG;;
+    *) echo "Error Occurred, Usage: SchemaSpy.sh -u [username] -p [password] -d [database] -h [host]" >&2
+       exit 1
+  esac
+done
+
+CMD="java -jar schemaSpy.jar -t pgsql -s bookbrainz "
+
+if [ ! -z "$HOST" ]
+then
+  CMD+="-host ${HOST} "
+else
+  throw_error "host: -h [host]"
+fi
+
+if [ ! -z "$DATABASE" ]
+then
+  CMD+="-db ${DATABASE} "
+else
+  throw error "database: -d [database]"
+fi
+
+if [ ! -z "$_USERNAME" ]
+then
+  CMD+="-u ${_USERNAME} "
+else
+  throw_error "username: -u [username]"
+fi
+
+CMD+="-p '${PASSWORD}' "
+CMD+="-o _build -dp postgresql-9.4-1202.jdbc41.jar;"
+
+eval $CMD
```

Comment:
You could also `echo $CMD` before the eval, to add some readability to the process.
We don't want to overdo it with logging, but I think that one line is a good compromise.